bql/planner/data_access.go

```diff
@@ -409,7 +409,12 @@ func simpleFetch(ctx context.Context, gs []storage.Graph, cls *semantic.GraphCla
 // table. The semantic graph clause is also passed to be able to identify what
 // bindings to set.
 func addTriples(ts <-chan *triple.Triple, cls *semantic.GraphClause, tbl *table.Table) error {
+	var lastErr error
 	for t := range ts {
+		if lastErr != nil {
+			// Drain the channel to avoid leaking goroutines.
+			continue
+		}
```

Comment:
**rbkloss** commented:

I think we can simplify this by always iterating on the ts channel after the loop which would effectively drain the channel.
It would be then your option of ignoring the error altogether or checking if it is non nil and breaking the loop to the draining one. Maybe even extract the channel draining logic to a DrainChannel function.

Specifically I think a break, instead of a continue will make readability easier.

Reply:
+1 . Not sure if this same as Ricardo's suggestion but indeed refactoring the 'draining'/skipping logic would indeed make the code more readable. In pseudocode we could have something like
for..
 ignore, err := IgnoreTriple()
 if ignore {
  continue
}
if err != nil {
  lastError = err
  continue
}
AddRow

Reply:
Thank you for the comment @jlsotomayorm. 
I have just mixed both of your comments to refactor the function `addTriples`, creating a new local boolean function `skipTriple` there and also separating the draining of the channel in a second loop below the original one as @rbkloss suggested.